api/src/main/java/org/apache/iceberg/encryption/EncryptionKeyMetadata.java

```diff
@@ -49,4 +54,12 @@ static EncryptionKeyMetadata empty() {
   ByteBuffer buffer();
 
   EncryptionKeyMetadata copy();
+
+  default ByteBuffer encryptionKey() {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  default ByteBuffer aadPrefix() {
+    return null;
```

Comment:
Do we handle a null `aadPrefix` or do we assume it is non-null?

Reply:
Null is technically possible. It'd be indeed safer to throw an unsupported exception here as well.